### ISSUE TITLE
merge perSystem system argument with the rest

### ISSUE THREAD

**Issue by zimbatm** [hercules-ci/flake-modules-core#18]
It's a small ergonomic thing but I think it would be best if perSystem would take one argument instead of two.

Current:
```
perSystem = system: { config, self', inputs', pkgs, ... }: {
```

Proposed change:
```
perSystem = { system, config, self', inputs', pkgs, ... }: {
```

What do you think?

**Comment by roberth** [comment]
I agree that this is a little odd, especially as `system` is usually irrelevant.

Maybe it should be a [`deferredModule`](https://github.com/NixOS/nixpkgs/pull/163617) but with options added to docs.

Fwiw, the reason for `system:` is that it reflects the structure of the options. `perSystem` is a `functionTo` `submodule`.

**Comment by zimbatm** [comment]
> especially as system is usually irrelevant.

and it's also available in `pkgs.system`

I don't understand the technical arguments around the module system :)
